RoddyCore/src/de/dkfz/roddy/Roddy.java

```diff
@@ -794,34 +794,73 @@ public static void loadPropertiesFile() {
 
         applicationProperties = new AppConfig(file);
 
-        getApplicationProperty("useRoddyVersion", LibrariesFactory.PLUGIN_VERSION_CURRENT); // Load some default properties
-        getApplicationProperty("usePluginVersion");
-        getApplicationProperty("pluginDirectories");
-        getApplicationProperty("configurationDirectories");
+        getOrSetApplicationProperty("useRoddyVersion", LibrariesFactory.PLUGIN_VERSION_CURRENT); // Load some default properties
+        getOrSetApplicationProperty("usePluginVersion");
+        getOrSetApplicationProperty("pluginDirectories");
+        getOrSetApplicationProperty("configurationDirectories");
     }
 
     public static AppConfig getApplicationConfiguration() {
         return applicationProperties;
     }
 
-    public static String getApplicationProperty(String pName) {
-        return getApplicationProperty(pName, "");
+    public static String getOrSetApplicationProperty(String pName) {
+        return getOrSetApplicationProperty(pName, "");
     }
 
-    public static String getApplicationProperty(RunMode runMode, String pName) {
-        return getApplicationProperty(runMode, pName, "");
+    public static String getOrSetApplicationProperty(RunMode runMode, String pName) {
+        return getOrSetApplicationProperty(runMode, pName, "");
     }
 
-    public static String getApplicationProperty(RunMode runMode, String pName, String defaultValue) {
-        return getApplicationProperty(runMode.name() + "." + pName, getApplicationProperty(pName, defaultValue));
+    public static String getOrSetApplicationProperty(RunMode runMode, String pName, String defaultValue) {
+        return getOrSetApplicationProperty(runMode.name() + "." + pName, getOrSetApplicationProperty(pName, defaultValue));
     }
 
-    public static String getApplicationProperty(String pName, String defaultValue) {
+    public static String getOrSetApplicationProperty(String pName, String defaultValue) {
```

Comment:
Do you think it makes sense to move the verification code below to the INI loader class?

Reply:
Yes, but thought I postpone this (scope and time). Now it's implemented anyway, although not without effort.